serverless/serverless
# Make `deploy function` lifecycle events more fine grain

## What did you implement:

Exposed new lifecycle events for `sls deploy function`

Currently there is no way to hook into the different phases of `sls deploy function`.  This prevents things like being able to browserify/webpack during `sls deploy function`.  This can be done on the `sls deploy` via the `deploy:createDeploymentArtifacts` hook, so this PR adds equivalent capabilities.
## How did you implement it:

Exposed 2 new `lifecycleEvents` to `deploy.function`
## How can we verify it:

Just run `sls deploy function -f <name>` and verify function deploys the same it did before this PR.

**_Is this ready for review?:**_ YES
